2-Authorization-I/1-call-graph/README.md

```diff
@@ -299,56 +309,53 @@ This sample app declares that it's CAE-capable by adding the `clientCapabilities
-    }
+ /**
+   * This method inspects the HTTPS response from a http call for the "www-authenticate header"
+   * If present, it grabs the claims challenge from the header and store it in the sessionStorage
+   * For more information, visit: https://docs.microsoft.com/en-us/azure/active-directory/develop/claims-challenge#claims-challenge-header-format
+   * @param response
+   */
+  handleClaimsChallenge(response: any, providerOptions: ProviderOptions): void {
+    const authenticateHeader: string = response.headers.get('www-authenticate');
+    const claimsChallengeMap: any = this.parseChallenges(authenticateHeader);
+    let account: AccountInfo = this.authService.instance.getActiveAccount()!;
+    addClaimsToStorage(
+      claimsChallengeMap.claims,
+      `cc.${msalConfig.auth.clientId}.${account?.idTokenClaims?.oid}.${
+        new URL(providerOptions.endpoint).hostname
+      }`
+    );
+    
+    new MsalAuthenticationProvider(providerOptions, this.authService).getAccessToken()
+    
   }
-```
 
-After that, we require a new access token via the `acquireTokenRedirect` API, fetch the claims challenge from the browser's localStorage, and pass it to the `acquireTokenRedirect` hook in the request parameter.
 
-```typescript
- getAccessTokenInteractively(): void {
-    this.authService.instance.acquireTokenRedirect({
-      account: this.authService.instance.getActiveAccount()!,
-      scopes: protectedResources.graphMe.scopes,
-      claims:
-        this.authService.instance.getActiveAccount()! &&
-        getClaimsFromStorage(
-          `cc.${msalConfig.auth.clientId}.${
-            this.authService.instance.getActiveAccount()?.idTokenClaims?.oid
-          }.${protectedResources.graphMe.resource}`
-        )
-          ? window.atob(
-              getClaimsFromStorage(
-                `cc.${msalConfig.auth.clientId}.${
-                  this.authService.instance.getActiveAccount()?.idTokenClaims
-                    ?.oid
-                }.${protectedResources.graphMe.resource}`
-              )
-            )
-          : '',
+  /**
+   * This method parses WWW-Authenticate authentication headers 
+   * @param header
+   * @return {Object} challengeMap
+   */
+  parseChallenges<T>(header: string): T {
+    const schemeSeparator = header.indexOf(' ');
+    const challenges = header.substring(schemeSeparator + 1).split(',');
+    const challengeMap = {} as any;
+
+    challenges.forEach((challenge: string) => {
+      const [key, value] = challenge.split('=');
+      challengeMap[key.trim()] = window.decodeURI(value.replace(/['"]+/g, ''));
     });
+
+    return challengeMap;
   }
+
 ```
 
+After that, we require a new access token via the `MsalAuthenticationProvider` Class, fetch the claims challenge from the browser's localStorage, and pass it to the `acquireToken` API in the request parameter. This is shown in [graph.service.ts](../SPA/src/app/graph.service.ts)
+
 ### Working with multiple resources
```

Comment:
From here on downwards a lot of text is being repeated in all samples. @derisen , @salman90 , consider moving them to the msaj.js wiki  and link them here?